src/Node/ReadLine.purs

```diff
@@ -1,42 +1,92 @@
 -- | This module provides a binding to the Node `readline` API.
 
-module Node.ReadLine where
+module Node.ReadLine
+  ( Interface
+  , READLINE
+  , InterfaceOptions
+  , Completer
+  , LineHandler
+  , createInterface
+  , createConsoleInterface
+  , noCompletion
+  , prompt
+  , setPrompt
+  , setLineHandler
+  , close
+  ) where
 
 import Prelude (return)
-
-import Control.Monad.Eff
-import Control.Monad.Eff.Console
+import Control.Monad.Eff (Eff)
+import Control.Monad.Eff.Console (CONSOLE)
+import Control.Monad.Eff.Exception (EXCEPTION)
+import Data.Maybe (Maybe(Just, Nothing), fromMaybe)
+import Node.Process (stdin, stdout)
+import Node.Stream (Readable, Writable)
 
 -- | A handle to a console interface.
 -- |
 -- | A handle can be created with the `createInterface` function.
 foreign import data Interface :: *
 
+-- | The effect of interacting with a stream via an `Interface`
+foreign import data READLINE :: !
+
+foreign import createInterfaceImpl :: forall opts eff. {| opts} -> Eff (readline :: READLINE | eff) Interface
```

Comment:
Or, rather, `assoc` is still there, but no longer needs the `IsOption` constraint, because the information that was previously available via that constraint (namely, how to turn a value of the type `value` into an `Options`) is now available on the `Option` itself.